Update config flow docs for read-only selectors (#2680)
home-assistant/developers.home-assistant: docs/data_entry_flow_index.md

        )
```

#### Displaying read-only information

> Some integrations have configuration which is frozen after initial configuration. When displaying an options flow, you can show this information in a read-only way, so that users may remember which options were selected during the initial configuration. To show this, define an optional selector in the same manner as the original selector, but add the `read_only=True` flag to its configuration. 

Reviewer: ```suggestion
Some integrations have options which are frozen after initial configuration. When displaying an options flow, you can show this information in a read-only way, so that users may remember which options were selected during the initial configuration. For this, define an optional selector as usual, but with the `read_only` flag set to `True`.
```

Also, is this only for Optional options, or also works with Required ones?

Author: The distinction between Optional and Required here is a bit meaningless, as no data is being returned for a read-only selector, it just sends the data to frontend to display, but nothing comes back. 

I believe it is just always defined as Optional in the schema, as describing a read-only selector as Required would complain that it wasn't provided by the user (which it never will be).  But it will certainly work to display data that was marked as Required in a previous flow step.
